### PR TITLE
Add Black pre-commit config

### PR DESCRIPTION
**Pull request by 3eak**
This will allow our code to be auto-formatted on every commit and will keep our codebase looking a bit tidier.

### CODE REVIEW

File: koala/settings.py [hunk 1]
```diff
@@ -23,72 +24,72 @@
...
 
 # SECURITY WARNING: keep the secret key used in production secret!
-PRODUCTION_SECRET = os.getenv('KOALA_SECRET_KEY')
-if (PRODUCTION_SECRET is not None):
+PRODUCTION_SECRET = os.getenv("KOALA_SECRET_KEY")
+if PRODUCTION_SECRET is not None:
     SECRET_KEY = PRODUCTION_SECRET
 else:
-    SECRET_KEY = 'n_-g*lp5k-uql%u=5c)m@4q2-o(ktesz8%qxap62*t4i0kt36!'
+    SECRET_KEY = "n_-g*lp5k-uql%u=5c)m@4q2-o(ktesz8%qxap62*t4i0kt36!"
```
Comment (JayDwee) [inline koala/settings.py:31]: Who's secret key is this?

**Review by JayDwee — COMMENTED** [review]
Sorry, don't know enough about this stuff to do a proper review :(
Had a quick look, but without knowing pre-commit stuff I can't help :(

File: koala/settings.py (hunk 1 above)
Comment (3eak) [reply koala/settings.py:31]: I think it's mine - there is another PR that hides this (and generates a new one)

**Comment by 3eak** [comment]
> Sorry, don't know enough about this stuff to do a proper review :(
> Had a quick look, but without knowing pre-commit stuff I can't help :(

Pre-commit is really awesome. It allows you to run jobs on a commit. You could even run a whole test suite on commits to ensure the code you're committing is safe.

In this example; every commit it runs a python formatter which will clean up the code on commits, and tell you if your code needed formatting - it will return your newly formatted file(s) unstaged and ready to be added and committed.

**Pull request closed without merging**